src/librustc_mir_build/hair/pattern/_match.rs

```diff
@@ -291,30 +291,39 @@ impl<'tcx> LiteralExpander<'tcx> {
                     }
                     Scalar::Raw { .. } => {
                         let layout = self.tcx.layout_of(self.param_env.and(rty)).unwrap();
-                        if layout.is_zst() {
-                            // Deref of a reference to a ZST is a nop.
-                            ConstValue::Scalar(Scalar::zst())
-                        } else {
-                            // FIXME(oli-obk): this is reachable for `const FOO: &&&u32 = &&&42;`
-                            bug!("cannot deref {:#?}, {} -> {}", val, crty, rty);
-                        }
+                        assert!(layout.is_zst());
+                        // Deref of a reference to a ZST is a nop.
+                        ConstValue::Scalar(Scalar::zst())
                     }
                 }
             }
-            // unsize array to slice if pattern is array but match value or other patterns are slice
-            (ConstValue::Scalar(Scalar::Ptr(p)), ty::Array(t, n), ty::Slice(u)) => {
+            // Unsize array to slice if pattern is array
+            // but match value or other patterns are slice.
```

Comment:
```suggestion
			// Unsize the array to a slice if we're matching on a slice,
			// or other patterns are a slice (despite this pattern being
			// an array).
```
I found the current comment a little hard to parse.